exes/button/src/main.cpp

```diff
@@ -0,0 +1,153 @@
+#include <cstdlib>
+#include <iostream>
+
+#include <boost/asio.hpp>
+#include <boost/program_options.hpp>
+#include <boost/sml.hpp>
+
+#include <tfc/ipc.hpp>
+#include <tfc/logger.hpp>
+#include <tfc/progbase.hpp>
+
+namespace asio = boost::asio;
+
+namespace sml = boost::sml;
+using sml::operator""_s;
+using sml::operator""_e;
+
+using std::chrono::milliseconds;
+using std::chrono_literals::operator""ms;
+using std::chrono::system_clock;
+
+// Events
+struct high {};
+struct low {};
+struct timeout {};
+
+struct my_deps {
+  tfc::ipc::bool_signal& long_signal;
+  tfc::ipc::bool_signal& double_signal;
+  tfc::ipc::bool_signal& touch_signal;
+  asio::io_context& ctx;
+  tfc::logger::logger& logger;
```

Comment:
how about owning these references?

Reply:
Sure, that looks better. ctx and logger kept as a reference.